script.sh

```diff
@@ -208,14 +210,51 @@ RUN echo "deb-src ${TRAVIS_DEBIAN_MIRROR} experimental main" >> /etc/apt/sources
 EOF
 fi
 
+case "${TRAVIS_DEBIAN_EXTRA_REPOSITORY}" in
+	https:*)
+		EXTRA_PACKAGES="apt-transport-https"
+		;;
+esac
+
+if [ -n "${TRAVIS_DEBIAN_EXTRA_REPOSITORY_GPG_URL}" ]
+then
+	EXTRA_PACKAGES="${EXTRA_PACKAGES} wget gnupg"
+fi
+
 cat >>Dockerfile <<EOF
 RUN apt-get update && apt-get dist-upgrade --yes
-RUN apt-get install --yes --no-install-recommends build-essential equivs devscripts git-buildpackage ca-certificates pristine-tar
+RUN apt-get install --yes --no-install-recommends build-essential equivs devscripts git-buildpackage ca-certificates pristine-tar ${EXTRA_PACKAGES}
 
 WORKDIR $(pwd)
 COPY . .
 EOF
 
+if [ -n "${TRAVIS_DEBIAN_EXTRA_REPOSITORY_GPG_URL}" ]
+then
+	cat >>Dockerfile <<EOF
+RUN wget -O- "${TRAVIS_DEBIAN_EXTRA_REPOSITORY_GPG_URL}" | apt-key add -
+EOF
+fi
+
+# We're adding the extra repository only after the essential tools have been
+# installed, so that we have apt-transport-https if the repository needs it.
+#
+# The subsequent apt-get update syncs only the content of the extra repository,
```

Comment:
Saves a couple of seconds on each build but it's not really necessary. If you don't mind that some package lists are downloaded twice (only when `$TRAVIS_DEBIAN_EXTRA_REPOSITORY` is set anyway), I will remove it.